SUPINT-1785: Describe elastic upgrade (#1604)
nuxeo/doc.nuxeo.com-content: src/nxdoc/nuxeo-server/administration/elasticsearch-setup.md

audit.elasticsearch.enabled=false
```

## Rebuilding the Repository Index{{> anchor 'reindex'}}

If you need to reindex the whole repository, you have different possibilities:

> ### Re-index the repository using the WorkManager (the legacy way)

Reviewer: ```suggestion
### Re-index the Repository Using the WorkManager (the Legacy Way)
```

 
There are 3 ways to run it:
 
1. you can do this from the [Nuxeo Dev Tool Browser Extension]({{page page='nuxeo-dev-tools-extension'}}#features).

2. you can do that from the JSF UI > Admin center > Elasticsearch > Admin

3. you can simply use `curl`

```bash

> curl -X POST "<NUXEO_URL>/nuxeo/site/automation/Elasticsearch.Index" -u Administrator:<PASSWORD> -H 'content-type: application/json+nxrequest' -d '{"params":{},"context":{}}'

Reviewer: `application/json+nxrequest` is deprecated, `application/json` should be used instead, and other occurrences below

    elasticsearch.reindex.bucketWriteSize=50
    ```

> ### Re-index the repository using the Bulk Service

Reviewer: ```suggestion
### Re-index the Repository Using the Bulk Service
```

Run a bulk command to re-index the repository, the command id is returned:
```bash

> curl -s -X POST "<SERVER_URL>/nuxeo/site/automation/Elasticsearch.BulkIndex" -u Administrator:<PASSWORD> -H 'content-type: application/json+nxrequest' -d '{"params":{},"context":{}}'

Reviewer 2: `application/json+nxrequest` is deprecated since 10.3 in favor of `application/json`.

{"commandId": "21aeaea1-0ef0-4a89-a92d-fa8f679361de"}
```

> Anytime you can request the status of the re-indexing using the previous command id:

Reviewer: ```suggestion
At any time, you can request the status of the re-indexing using the previous command id:
```
